src/consumer.rs

```diff
@@ -186,6 +186,7 @@ impl ConsumerInner {
 
     fn cancel(&mut self) -> Result<()> {
         trace!("cancel; consumer_tag={}", self.tag);
+        self.drop_deliveries();
```

Comment:
Any reason to move this earlier?

Reply:
Yes, it's actually important. `drop_deliveries` cleans the consumer channel. Unfortunately `Ok(None)` is also being purged and in the stream polling, this leads to `delivery; status=NotReady`.